error/s2n_errno.h

```diff
@@ -175,6 +175,7 @@ typedef enum {
     S2N_ERR_ARRAY_INDEX_OOB,
     S2N_ERR_FREE_STATIC_BLOB,
     S2N_ERR_RESIZE_STATIC_BLOB,
+    S2N_ERR_PROTOCOL_DOWNGRADE_DETECTED,
```

Comment:
nit, could you move this to the S2N_ERR_T_PROTO category? also try moving the ERR_ENTRY to the relevant order.

Reply:
I chose S2N_ERR_T_INTERNAL because this error seemed very close to S2N_ERR_{CLIENT,SERVER}_PROTOCOL_VERSION error. But I could see it being in the same category as S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED as well.